src/lib.rs

```diff
@@ -89,6 +91,14 @@ where
         Ok(bytes[0] == who_am_i)
     }
 
+    pub fn whoami_ag(&mut self) -> Result<u8, T::Error> {
+        //let mut bytes = [0u8; 1];
+
+        let mut id = self.read_register(Sensor::Accelerometer, register::AG::WHO_AM_I.addr())?;
```

Comment:
This doesn't need to be mutable. You can remove `mut`.

Reply:
Yes, `let mut id`. It doesn't have to be mutable. 
And you probably want to delete line 95, `//let mut bytes = [0u8; 1];` as it is not doing anything.

Reply:
OK, fixed that.